dGame/dUtilities/Mail.cpp

```diff
@@ -397,7 +398,8 @@ void Mail::HandleMailRead(RakNet::BitStream* packet, const SystemAddress& sysAdd
 }
 
 void Mail::HandleNotificationRequest(const SystemAddress& sysAddr, uint32_t objectID) {
-	std::async(std::launch::async, [&]() {
+	// We need to assign the result of this async to a variable so the thread here isnt blocking.
+	auto futureValue = std::async(std::launch::async, [&]() {
```

Comment:
FYI: I don't think this variable improves things much. The issue is that `std::async  returns a `std::future` whose destructor will wait for the thread/task to complete according to the box in <https://en.cppreference.com/w/cpp/thread/future/%7Efuture> (seems to be a C++ 14 thing?)

Reply:
thats fair.  I didnt mean to send those to this branch for what its worth.  Meant for this to go into its own branch.  Ill address that tomorrow.  Ill also look up a way to make this properly async on all platforms.